plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeSystemTables.java

```diff
@@ -103,4 +112,339 @@ public void testPropertiesTable()
+
+        assertUpdate("INSERT INTO " + tableName + " VALUES (1, 1, 'ala'), (2, 2, 'kota'), (3, 3, 'osla')", 3);
+        assertThat(query("SELECT CAST(data.id AS ROW(INTEGER, INTEGER, BIGINT)) FROM \"" + tableName + "$partitions\""))
+            .matches("VALUES ROW(ROW(2, 200, CAST(0 AS BIGINT))), ROW(ROW(3, 300, CAST(0 AS BIGINT))), ROW(ROW(1, 100, CAST(0 AS BIGINT)))");
+    }
+
+    @Test
+    public void testColumnMappingModePartitionsTable()
+    {
+        for (String columnMappingMode : ImmutableList.of("id", "name", "none")) {
+            String tableName = "test_simple_column_mapping_mode_" + columnMappingMode + "_partitions_table";
+            try {
+                assertUpdate("CREATE TABLE " + tableName + " (_bigint BIGINT, _date DATE) WITH (column_mapping_mode = '" + columnMappingMode + "', partitioned_by = ARRAY['_date'] )");
+                assertUpdate("INSERT INTO " + tableName + " VALUES (0, CAST('2019-09-08' AS DATE)), (1, CAST('2019-09-09' AS DATE)), (2, CAST('2019-09-09' AS DATE))", 3);
+                assertUpdate("INSERT INTO " + tableName + " VALUES (3, CAST('2019-09-09' AS DATE)), (4, CAST('2019-09-10' AS DATE)), (5, CAST('2019-09-10' AS DATE))", 3);
+                assertUpdate("INSERT INTO " + tableName + " VALUES (6, NULL)", 1);
+
+                assertQuerySucceeds("SELECT * FROM " + tableName);
+                assertQuery("SELECT count(*) FROM " + tableName, "VALUES 7");
+                assertQuery("SELECT count(*) FROM \"" + tableName + "$partitions\"", "VALUES 4");
+
+                assertQuery("SHOW COLUMNS FROM \"" + tableName + "$partitions\"",
+                        """
+                                VALUES
+                                ('partition', 'row(_date date)', '', ''),
+                                ('file_count', 'bigint', '', ''),
+                                ('total_size', 'bigint', '', ''),
+                                ('data', 'row(_bigint row(min bigint, max bigint, null_count bigint))', '', '')
+                                """);
+
+                assertQuery(
+                    "SELECT partition._date FROM \"" + tableName + "$partitions\"", " VALUES DATE '2019-09-08', DATE '2019-09-09', DATE '2019-09-10', NULL");
+
+                assertThat(query("SELECT CAST(data._bigint AS ROW(BIGINT, BIGINT, BIGINT)) FROM \"" + tableName + "$partitions\""))
+                    .matches("""
+                        VALUES 
+                        ROW(ROW(CAST(1 AS BIGINT), CAST(3 AS BIGINT), CAST(0 AS BIGINT))), 
+                        ROW(ROW(CAST(0 AS BIGINT), CAST(0 AS BIGINT), CAST(0 AS BIGINT))), 
+                        ROW(ROW(CAST(6 AS BIGINT), CAST(6 AS BIGINT), CAST(0 AS BIGINT))),
+                        ROW(ROW(CAST(4 AS BIGINT), CAST(5 AS BIGINT), CAST(0 AS BIGINT)))
+                        """);
+            }
+            finally{
+                assertUpdate("DROP TABLE IF EXISTS " + tableName);
+            }
```

Comment:
extract this code to a helper method and make a separate call for each : none, id , name